src/OSPSuite.Core/Domain/Services/ConcurrencyManager.cs

```diff
@@ -70,7 +71,8 @@ public class ConcurrencyManager : IConcurrencyManager
          numberOfCoresToUse = Math.Min(numberOfCoresToUse, concurrentData.Count);
 
          var results = new ConcurrentDictionary<TData, ConcurrencyManagerResult<TResult>>();
-         //Starts one task per core
+
+         //Starts one task per core <= THIS IS NOT QUITE RIGHT I BELIEVE
```

Comment:
@abdelr This does not start a task per core. It await an existing task./
Would you mind explaining what this code is doing.  I am not sure that it is actually splitting on all available cores. Can we chat?